examples/notebooks/Splitters Tutorial.ipynb

```diff
@@ -0,0 +1,370 @@
+      "TIMING: featurizing shard 0 took 0.029 s\n",
+      "TIMING: dataset construction took 0.038 s\n",
+      "Loading dataset from disk.\n"
+     ]
+    }
+   ],
+   "source": [
+    "from deepchem.splits.splitters import SpecifiedSplitter\n",
+    "\n",
+    "input_file=os.path.join(\"/home/ab/deepchem/deepchem/models/tests/user_specified_example.csv\")\n",
+    "\n",
+    "tasks=['log-solubility']\n",
+    "featurizer=dc.feat.CircularFingerprint(size=1024)\n",
+    "loader = dc.data.CSVLoader(tasks=tasks, smiles_field=\"smiles\",featurizer=featurizer)\n",
+    "dataset=loader.featurize(input_file)\n",
+    "\n",
+    "split_field='split'\n",
+    "\n",
+    "splitter=SpecifiedSplitter(input_file,split_field)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data,valid_data,test_data=splitter.split(dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "When we split the data using the specified splitter it compares the data in each row of the `split_field` with train,test and valid and then splits the data according to the data in the `split_field`."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 30,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "([0, 1, 2, 3, 4, 5], [6, 7], [8, 9])"
+      ]
+     },
+     "execution_count": 30,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "train_data,valid_data,test_data"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Another splitter present in the fraework is `IndiceSplitter`. This splitter takes an input of valid_indices and test_indices which are lists with the indices of validation data and test data in the dataset respectively."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 36,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from deepchem.splits.splitters import IndiceSplitter\n",
+    "\n",
+    "splitter=IndiceSplitter(valid_indices=[7],test_indices=[9])"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 37,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "([0, 1, 2, 3, 4, 5, 6, 8], [7, 9], [9])"
+      ]
+     },
+     "execution_count": 37,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "splitter.split(dataset)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The splitter which can be used to split the data on the basis of groupings is the `RandomGroupSplitter`. This splitter that splits on groupings. \n",
+    "\n",
+    "An example use case is when there are multiple conformations of the same molecule that share the same topology.This splitter subsequently guarantees that resulting splits preserve groupings.\n",
```

Comment:
This doesn't show the power of RandomGroupSplitter.  Put two elements in a group and then show they are in the same group.